src/SFA.DAS.EAS.Portal.UnitTests/Application/Commands/Cohort/CohortApprovalRequestedCommandHandlerTests.cs

```diff
@@ -0,0 +1,220 @@
+                // arrange
+                var testContext = new TestContext();
+                string cohortReference = Guid.NewGuid().ToString();
+                Types.Cohort cohort = new CohortBuilder().WithId(cohortReference);
+                testContext.TestAccount.Organisations.First().Cohorts.Add(cohort);
+                testContext.TestCommitment.Reference = cohortReference;
+                testContext.TestAccount.Organisations.First().Cohorts.Count.Should().Be(1);
+
+                CohortApprovalRequestedCommand command = new CohortApprovalRequestedCommandBuilder();
+
+                // act
+                await testContext.Sut.Handle(command);
+
+                //assert
+                testContext.MockAccountsService.Verify(m =>
+                m.Save(It.Is<AccountDocument>(a =>
+                a.Account.Organisations.First().Cohorts.Count.Equals(1)), It.IsAny<CancellationToken>()),
+                Times.Once);
+            }
+
+
+            [Test]
+            public async Task WhenANewCohortIsAdded_ThenApprentishipsInTheCohortAreStored()
+            {
+                // arrange
+                var testContext = new TestContext();
+                string cohortReference = Guid.NewGuid().ToString();
+                long apprenticeshipId = 123;
+
+                testContext.TestCommitment.Reference = cohortReference;
+                testContext.TestCommitment.Apprenticeships = new List<Commitments.Api.Types.Apprenticeship.Apprenticeship>() { new Commitments.Api.Types.Apprenticeship.Apprenticeship { Id = apprenticeshipId } };
+
+                CohortApprovalRequestedCommand command = new CohortApprovalRequestedCommandBuilder();
+
+                // act
+                await testContext.Sut.Handle(command);
+
+                //assert
+                testContext.MockAccountsService.Verify(m =>
+                m.Save(It.Is<AccountDocument>(a =>
+                a.Account.Organisations.First().Cohorts.Count.Equals(1) &&
+                a.Account.Organisations.First().Cohorts.First().Apprenticeships.Count.Equals(1) &&
+                a.Account.Organisations.First().Cohorts.First().Apprenticeships.First().Id.Equals(apprenticeshipId)), It.IsAny<CancellationToken>()),
+                Times.Once);
+            }
+
+            [Test]
+            public async Task WhenAnExistingCohortIsHandled_ThenAnyApprentishipsChangesAreAlsoStored()
+            {
+                // arrange
+                var testContext = new TestContext();
+                string cohortReference = Guid.NewGuid().ToString();
+                Apprenticeship apprenticeship = new ApprenticeshipBuilder();
+                Types.Cohort cohort = new CohortBuilder()
+                    .WithId(cohortReference)
+                    .WithApprenticeship(apprenticeship);
+                testContext.TestAccount.Organisations.First().Cohorts.Add(cohort);
+                testContext.TestAccount.Organisations.First().Cohorts.Count.Should().Be(1);
```

Comment:
Why are we asserting in the arrange phase?

Reply:
Checking the data is consistent before performing the test.  Someone could change the builders and this test wouldn't know about it.  It's an edge case that shouldn't happen.